src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -106,6 +106,9 @@
     <Error Condition="'$(RunAOTCompilation)' == 'true' and '$(_IsEMSDKMissing)' == 'true'"
            Text="$(_EMSDKMissingErrorMessage) Emscripten SDK is required for AOT'ing assemblies." />
 
+    <Error Condition="'$(RunAOTCompilation)' == 'true' and '$(EnableAggressiveTrimming)' != 'true'"
+           Text="EnableAggressiveTrimming=true is required for AOT'ing assemblies." />
```

Comment:
This property is only for runtime test projects. Use `$(PublishTrimmed)` instead. Also, we have a couple of tests in WBT that AOT with `PublishTrimmed=false`.

https://github.com/dotnet/runtime/blob/0e12347861940389153ee2233ff398d551f86711/src/mono/wasm/Wasm.Build.Tests/NativeBuildTests.cs#L45-L68

https://github.com/dotnet/runtime/blob/0e12347861940389153ee2233ff398d551f86711/src/mono/wasm/Wasm.Build.Tests/NativeBuildTests.cs#L104-L135

Also, this will get called even for relinking during the `build` (!publish), and a user might have the two properties set to true in their project. So, also check for `WasmBuildingForNestedPublish==true`.

Reply:
yeah PublishTrimmed is really the user facing property EnableAgressiveTrimming is more for CI and tests